### PR TITLE
CA-913: fix(search): send all selected owners to the project search filter_by_owners param

### PR DESCRIPTION
**Pull request by Bazabizi**
## Summary

[CA-913](https://ripplearc.youtrack.cloud/issue/CA-913) — the project-search Owner sheet is multi-select and the `global_search` RPC already accepts `filter_by_owners uuid[]`, but `ProjectSearchBloc` truncated the selection to the alphabetically-first owner and the datasource wrapped that single value in a list. This widens the app-side contract so every selected owner reaches the RPC. App-only — no backend change (contrast CA-846, the tag param, which needs a backend signature migration).

Stacked on #486 (CA-901). Replaces the never-refiled contract-widening half of the superseded CA-771.

## Changes

`filterByOwner: String?` → `filterByOwners: List<String>?` through the full chain:
- `ProjectSearchRepository` (interface + impl), `ProjectSearchDataSource` (interface + `RemoteProjectSearchDataSource`), `FakeProjectSearchRepository`.
- `RemoteProjectSearchDataSource` sends the list directly (null when empty) instead of `[singleOwner]`.
- `ProjectSearchBloc._executeSearch` sends `_selectedOwnerIds` sorted (deterministic payload); the owner-truncation warning and CA-771 comments are removed. Tag truncation stays (CA-846) with its comment repointed.

## Tests

- Datasource test: asserts all owners reach `filter_by_owners`, plus a new empty-selection → `null` case.
- Repository-impl test: multi-owner array passthrough.
- Bloc test "threads all selected owners into the search RPC (sorted)" updated from the old truncation assertion; the CA-901 single-owner re-run test still asserts `['owner-1']`.
- analyze clean; custom_lint only the pre-existing `fake_router.dart` hit.

## Notes
- The unbound dead dashboard `RemoteProjectSearchDataSource` keeps the old `filterByOwner` signature — it's slated for deletion in CA-845 and has no live callers, so it's left untouched here.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: lib/libraries/project/testing/fake_project_search_repository.dart [hunk 1]
```diff
@@ -73,7 +73,7 @@ class FakeProjectSearchRepository implements ProjectSearchRepository {
     DateTime? filterByDateFrom,
     DateTime? filterByDateTo,
     String? filterByTag,
-    String? filterByOwner,
+    List<String>? filterByOwners,
   }) async {
```
Comment (ayanasamuel8) [inline lib/libraries/project/testing/fake_project_search_repository.dart:77]: `FakeProjectSearchRepository` is correctly updated here for interface compliance, but has zero callers in the codebase — no test or production file imports it. Pre-existing, not introduced by this PR. Worth opening a chore ticket to either wire it into a test or delete it before it accumulates more silent drift as interfaces evolve.

**Review by ayanasamuel8 — APPROVED** [review]
## Review: CA-913

Clean, well-scoped fix. All six production layers updated atomically with no partial rename left behind. One inline note on a pre-existing issue surfaced by the diff; one suggestion here.

### 🔵 Suggestion — E2E verification
The ticket AC requires "Multi-owner selection verified E2E on the local stack (two owners → union of their projects)". Please confirm this was run locally before merging.

### ✅ Positive findings
- **Full-chain atomicity** — `filterByOwner\b` returns zero production hits after this merge; no partial rename.
- **Defensive double-null guard** — `(filterByOwners == null || filterByOwners.isEmpty) ? null : filterByOwners` handles callers that pass empty instead of null, even though the BLoC already normalises this.
- **Sorted payload** — `(_selectedOwnerIds.toList()..sort())` mirrors the tag pattern; test validates sort order (`{'owner-2', 'owner-1'}` → `['owner-1', 'owner-2']`).
- **Owner truncation warning correctly removed** — bug is fixed so the `AppLogger.warning` is gone. Tag warning correctly kept (CA-846 pending).
- **New edge-case test** — `'sends a null owner array when no owners are selected'` pins the empty-list → null boundary at the data source level.
- **Test double pattern clean** — no mocks; all tests verify RPC params through `FakeSupabaseWrapper`.
- **Scope discipline** — global search (CA-848), tag array (CA-846), and dead dashboard data source (CA-845) correctly left alone with accurate explanations.

### 🎯 AC check
| AC | Status |
|---|---|
| `ProjectSearchBloc` sends full set of selected owner ids; truncation warning and CA-771 comments removed | ✅ |
| Bloc tests cover multi-owner param passing; datasource contract-pin test updated | ✅ |
| Global search owner path out of scope (CA-848) | ✅ |
| Multi-owner E2E verified on local stack | ⚠️ Author to confirm |

**Review by ripplearcgit — APPROVED** [review]
### Architectural overview

```mermaid
sequenceDiagram
    participant User
    participant Sheet as OwnerFilterSheet
    participant Bloc as ProjectSearchBloc
    participant Repo as ProjectSearchRepositoryImpl
    participant DS as RemoteProjectSearchDataSource
    participant RPC as global_search RPC
    User->>Sheet: select owner-1 and owner-2 multi-select
    Sheet->>Bloc: ProjectSearchOwnerFiltersAppliedEvent ownerIds
    Bloc->>Bloc: _selectedOwnerIds = Set.unmodifiable(ownerIds)
    Bloc->>Bloc: _executeSearch triggered
    alt _selectedOwnerIds is empty
        Bloc->>Repo: searchProjects filterByOwners null
    else _selectedOwnerIds has one or more owners
        Bloc->>Bloc: sort _selectedOwnerIds.toList()
        Bloc->>Repo: searchProjects filterByOwners sorted list
    end
    Repo->>DS: fetchProjectsBySearchQuery filterByOwners
    alt filterByOwners is null or empty
        DS->>RPC: rpc filter_by_owners null
    else filterByOwners has entries
        DS->>RPC: rpc filter_by_owners owner array
    end
    RPC-->>DS: projects union of both owners
    DS-->>Repo: List of ProjectDto
    Repo-->>Bloc: Right list of Project
```

**Verdict: CONDITIONAL LGTM.** No blockers — the `filterByOwner` → `filterByOwners` contract widening is atomic across the full chain (BLoC → repository interface/impl → data source interface/impl → fake), the sort-for-determinism behavior mirrors the existing tag-filter pattern, and both branches of the data source's null/empty guard are covered by tests. The dashboard-scoped `RemoteProjectSearchDataSource` in `lib/features/dashboard/data/data_source/` is a different class implementing a different interface and is confirmed dead (no DI registration, no live importer) — correctly left untouched, tracked separately in CA-845.

### Nits

🟡 **Confirm multi-owner E2E was run locally before merge**

Carried forward from the existing approving review by Ayana on this same HEAD SHA (`f2b3eef`, unchanged): please confirm multi-owner E2E was run on the local stack before merge (two owners selected → search returns the union of their projects), per the ticket's acceptance criteria. Not verifiable from the diff alone, so this is not a code-line finding — just a pre-merge confirmation ask. Non-blocking.

Full HTML report: ~/Desktop/pr-reviews/pr-review-construculator-app-487.html

File: lib/libraries/project/testing/fake_project_search_repository.dart (hunk 1 above)
Comment (Bazabizi) [reply lib/libraries/project/testing/fake_project_search_repository.dart:77]: Filed as [CA-953](https://ripplearc.youtrack.cloud/issue/CA-953): wire `FakeProjectSearchRepository` into a real test consumer or delete it, so it stops accumulating compile-only drift on interface changes.
